coder_sniffer/DrupalPractice/Sniffs/Objects/StrictSchemaDisabledSniff.php

```diff
@@ -0,0 +1,135 @@
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+namespace DrupalPractice\Sniffs\Objects;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\AbstractVariableSniff;
+
+/**
+ * Checks that $strictConfigSchema is not set to FALSE in test classes.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+class StrictSchemaDisabledSniff extends AbstractVariableSniff
+{
+
+    /**
+     * The name of the variable in the test base class to disable config schema checking.
+     */
+    const STRICT_CONFIG_SCHEMA_NAME = '$strictConfigSchema';
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function processMemberVar(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        if (($tokens[$stackPtr]['content'] === static::STRICT_CONFIG_SCHEMA_NAME) && ($this->isTestClass($phpcsFile, $stackPtr) === true)) {
+            $find = array(
+                     T_FALSE,
+                     T_TRUE,
+                     T_NULL,
+                     T_SEMICOLON,
+                    );
+            $next = $phpcsFile->findNext($find, ($stackPtr + 1));
+            // If this variable is being set, the only allowed value is TRUE.
+            // Otherwise if FALSE or NULL, schema checking is disabled.
+            if ($tokens[$next]['code'] !== T_TRUE) {
+                $error = 'Do not disable strict config schema checking in tests. Instead ensure your module properly declares its schema for configurations.';
+                $data  = [$tokens[$stackPtr]['content']];
+                $phpcsFile->addError(
+                    $error,
+                    $stackPtr,
+                    'StrictConfigSchema',
+                    $data
+                );
+            }
+        }//end if
+
+    }//end processMemberVar()
+
+
+    /**
+     * Determine if this class is a test class.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return bool
+     *   Returns TRUE if the current class is a test class.
+     */
+    protected function isTestClass(File $phpcsFile, $stackPtr)
+    {
+        // Only applies to test classes, which have Test in the name.
+        $types  = array(
+                   T_CLASS,
+                   T_TRAIT,
+                  );
+        $class  = $phpcsFile->findPrevious($types, ($stackPtr - 1));
```

Comment:
Instead of searching we can use something like `$classPtr    = key($tokens[$stackPtr]['conditions']);` as in ValidVariableNameSniff.

Reply:
I changed this to 

```
        // Only applies to test classes, which have Test in the name.
        $tokens   = $phpcsFile->getTokens();
        $classPtr = key($tokens[$stackPtr]['conditions']);
        $name     = $phpcsFile->findNext([T_STRING], $classPtr);
        return strpos($tokens[$name]['content'], 'Test') !== false;
```